src/main/java/com/aws/greengrass/logmanager/model/LogFile.java

```diff
@@ -107,4 +115,12 @@ public boolean isEmpty() {
         return Utils.isEmpty(this.hashString());
     }
 
+    /**
+     * Returns the source path of the file original file.
+     */
+    public String getSourcePath() {
```

Comment:
Let's not expose this. The fact that we're using hard links under the hood should be hidden as much as possible. Deletion is a special case. Let's handle that separately by exposing a delete method which can encapsulate the logic to handle rotation.

Reply:
+1

Reply:
Could we fix this one a separate PR? - We still need to fix the delete anyways